remove unused field for chat_template.default for DPO training

timofey opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

# Description

DPO trainer expects only prompt+chosen+rejected columns present in the final dataset. However, the `chat_template.default` implementation passes an extra column `messages` which causes validations to fail.

## Motivation and Context

Fixes issue #2415 

## How has this been tested?

Tested using sample config:

```yaml
rl: dpo

datasets:
  - path: "..../dataset.jsonl"
    chat_template: tokenizer_default
    type: chat_template.default
    field_messages: "messages"
    field_chosen: "chosen"
    field_rejected: "rejected"
    message_property_mappings:
      role: role
      content: content
    roles:
      user: [ "user" ]
      assistant: [ "assistant" ]
      system: [ "system" ]
```

then by running `axolotl preprocess` and `axolotl train` commands.

Without the fix, it fails with the following errors:

```
KeyError: "Invalid keys in the example: {'messages', 'prompt', 'rejected', 'chosen'}"
```

## Screenshots (if appropriate)

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

## Social Handles (Optional)

<!-- Thanks for submitting a bugfix or enhancement. -->
<!-- We'd love to show our thanks to you on Twitter & Discord if you provide your handle -->


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - The system now provides additional metadata indicating which columns should be removed during processing.
- **Bug Fixes**
  - Updated dataset tests to reflect correct feature presence and absence, improving data validation.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

timofey committed:
remove unused field for chat_template.default

"messages" field present in final dataset causes issues with DPO
training otherwise

winglian reviewed (changes requested):
The caller of the transform_fn is going to need to unpack the updated return (tuple)

winglian committed:
lint and fix tests for new return value

winglian commented:
I think I fixed the linting issues and tests that use the updated return values

winglian committed:
remove unused field for chat_template.default

"messages" field present in final dataset causes issues with DPO
training otherwise

lint and fix tests for new return value

fix for updated expected fields for dpo

remove unused field for chat_template.default

"messages" field present in final dataset causes issues with DPO
training otherwise

fix test still expecting "messages" field

timofey commented:
oh thanks @winglian ! I'll take a stab at it tonight as well, looks like some of the tests are still failing

timofey commented:
@winglian can you run the tests on the pipeline again please? Adjusted the failed test, since the "conversation" field is removed now

winglian committed:
chore: lint
